### PR TITLE
feat: add tcp host and port range configuration

### PR DESCRIPTION
**Pull request by michalg91**
## Describe the pull request
This PR adds posibility to configure port range on tcp port forwarding also has ability to set different hostname for tcp service (eg. for usage with kubernetes loadbalancer service)

Link to the issue: n/a

## Consent

- [x] I agree to follow the [Code of Conduct](https://go.dev/conduct) by submitting this pull request.
- [x] I have read and acknowledged the [Contributing guide](https://github.com/pgrok/pgrok/blob/main/.github/contributing.md).
- [ ] I have added test cases to cover the new code or have provided the test plan.

### CODE REVIEW

File: internal/conf/conf.go [hunk 1]
```diff
@@ -25,6 +26,11 @@ type Proxy struct {
 	Port   int    `yaml:"port"`
 	Scheme string `yaml:"scheme"`
 	Domain string `yaml:"domain"`
+	Tcp    struct {
```
Comment (unknwon) [inline internal/conf/conf.go:29]:
```suggestion
	TCP    struct {
```

File: internal/conf/conf.go [hunk 2]
```diff
@@ -57,6 +63,7 @@ func Load(configPath string) (*Config, error) {
 	}
 
 	var config Config
+	defaults.Set(&config)
```
Comment (unknwon) [inline internal/conf/conf.go:66]: I don't think it's worth using a package for setting defaults for two vars, let's do it manually after line 70.

**Review by unknwon — CHANGES_REQUESTED** [review]
Thanks for the PR!

In addition to the code change, we also need to update example YAML config https://github.com/pgrok/pgrok/blob/main/pgrokd.exmaple.yml

File: internal/conf/conf.go (hunk 2 above)
Comment (michalg91) [reply internal/conf/conf.go:66]: You're right, but maybe it's worth implementing to cover case with passing minimal yaml config in the future?

Comment (unknwon) [reply internal/conf/conf.go:66]: Yeah, ideally, we should be able to embedded a minimal YAML config, then overwrite values from the provided YAML.

Comment (michalg91) [reply internal/conf/conf.go]: I decided to create another PR for this one, right now i have assigned them in code.

File: internal/conf/conf.go [hunk 3]
```diff
@@ -57,6 +62,15 @@ func Load(configPath string) (*Config, error) {
 	}
 
 	var config Config
+
+	if config.Proxy.TCP.PortStart == 0 {
```
Comment (unknwon) [inline internal/conf/conf.go:66]:
```suggestion
	if config.Proxy.TCP.PortStart <= 0 {
```

File: internal/conf/conf.go [hunk 4]
```diff
@@ -57,6 +62,15 @@ func Load(configPath string) (*Config, error) {
...
+		config.Proxy.TCP.PortStart = 10000
+	}
+
+	if config.Proxy.TCP.PortEnd == 0 {
```
Comment (unknwon) [inline internal/conf/conf.go:70]:
```suggestion
	if config.Proxy.TCP.PortEnd <= 0 {
```

File: internal/sshd/client.go [hunk 5]
```diff
@@ -242,7 +242,10 @@ func (c *Client) handleServerInfo(proxy conf.Proxy, req *ssh.Request) {
 	var hostURL string
 	switch c.protocol {
 	case "tcp":
-		host := proxy.Domain
+		host := proxy.TCP.Domain
+		if host == "" {
+			host = proxy.Domain
+		}
```
Comment (unknwon) [inline internal/sshd/client.go:248]: We have this nice helper available 😁 

https://github.com/pgrok/pgrok/blob/eed3ebf3d483231f4177c8b6f20f8beb08f10990/internal/strutil/strutil.go#L44-L45

File: internal/sshd/client.go [hunk 6]
```diff
@@ -47,6 +47,7 @@ func (c *Client) handleHint(req *ssh.Request) {
 
 func (c *Client) handleTCPIPForward(
 	ctx context.Context,
+	proxy conf.Proxy,
 	cancel context.CancelFunc,
```
Comment (unknwon) [inline internal/sshd/client.go:51]: `ctx` and `cancel` are a pair, so let's put `proxy` in the third place of arguments.

File: pgrokd.exmaple.yml [hunk 7]
```diff
@@ -13,6 +13,14 @@ proxy:
   scheme: "http"
   # The domain name of the proxy URLs.
   domain: "localhost:3000"
+  # Optional TCP Proxy settings
```
Comment (unknwon) [inline pgrokd.exmaple.yml:16]:
```suggestion
  # Optional settings for the TCP proxy.
```

File: pgrokd.exmaple.yml [hunk 8]
```diff
@@ -13,6 +13,14 @@ proxy:
...
+  tcp:
+    # The domain name of the proxy URLs - defaults to proxy.domain value
+    domain: "localhost"
+    # TCP proxy start port
```
Comment (unknwon) [inline pgrokd.exmaple.yml:20]:
```suggestion
    # TCP proxy start port (inclusive)
```

File: pgrokd.exmaple.yml [hunk 9]
```diff
@@ -13,6 +13,14 @@ proxy:
...
+    port_start: 10000
+    # TCP proxy end port
```
Comment (unknwon) [inline pgrokd.exmaple.yml:22]:
```suggestion
    # TCP proxy end port (exclusive)
```

**Review by unknwon — CHANGES_REQUESTED** [review]
Thanks for the follow-up! Few last minor issues.

File: internal/sshd/client.go (hunk 5 above)
Comment (michalg91) [reply internal/sshd/client.go]: thanks!

**Review by unknwon — APPROVED** [review]
🚀

**Comment by unknwon** [comment]
This change is included in https://github.com/pgrok/pgrok/releases/tag/v1.4.1

**Pull request merged**